### PR TITLE
created test for database integration

### PR DESCRIPTION
**Pull request by Chenxi96**
updated package json

updated workflow and package json

updated again

commited

updated

commit issue

changed test yaml

update package

commit

final edit to test database integration

Update tests.yml

Update tests.yml

Update tests.yml

Update tests.yml

Update tests.yml

Update tests.yml

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests. (If tests are not applicable, explain why in the next section.)
- [x] Is there only 1 commit.

## Describe your changes
1. created test for database integration test.
2. After multiple attempts to have test pass through Github actions, I finally figured out how to have the test pass.

### CODE REVIEW

File: .env.test [hunk 1]
```diff
@@ -0,0 +1 @@
+POSTGRES_PRISMA_URL="postgresql://postgres:password@localhost:5432/court-share?schema=public"
```
Comment (Chenxi96) [inline .env.test:1]: This is used locally, it swaps the .env with this file to connect into the test database.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -5,11 +5,26 @@ on: [push, pull_request]
 jobs:
   test:
     runs-on: ubuntu-latest
+    services:
```
Comment (Chenxi96) [inline .github/workflows/tests.yml:8]: This creates a Postgres container to spin up a database to test the database integration.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -5,11 +5,26 @@ on: [push, pull_request]
...
     steps:
       - uses: actions/checkout@v2
       - name: Test using Node.js
         uses: actions/setup-node@v2
         with:
           node-version: '18'
       - run: npm install
-      - run: npm run test:ci
\ No newline at end of file
+      - name: Wait for PostgreSQL
```
Comment (Chenxi96) [inline .github/workflows/tests.yml:24]: This is going to wait for the Postgres database to be ready to use.

File: __tests__/lib/createPost.test.jsx
Comment (Chenxi96) [inline __tests__/lib/createPost.test.jsx:1]: Created test to test database integration

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -5,11 +5,26 @@ on: [push, pull_request]
 jobs:
   test:
     runs-on: ubuntu-latest
```
Comment (alvinc21) [inline .github/workflows/tests.yml:7]: Can you remind me if the hosted environment also runs on ubuntu?

**Review by alvinc21 — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 4 above)
Comment (Chenxi96) [reply .github/workflows/tests.yml:7]: Yeah i believe so

**Pull request merged**